pkg/store/limiter.go

```diff
@@ -135,10 +158,25 @@ func NewLimitedStoreServer(store storepb.StoreServer, reg prometheus.Registerer,
 			Name: "thanos_store_selects_dropped_total",
 			Help: "Number of select queries that were dropped due to configured limits.",
 		}, []string{"reason"}),
+		pendingRequestsGauge: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "thanos_store_server_pending_series_requests",
+			Help: "Number of pending series requests",
+		}),
+		maxPendingRequestLimitHit: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: "thanos_store_server_hit_max_pending_series_requests_total",
+			Help: "Number of pending series requests that hit the max pending request limit",
+		}),
+		maxPendingRequests: opts.MaxPendingSeriesRequests,
 	}
 }
 
 func (s *limitedStoreServer) Series(req *storepb.SeriesRequest, srv storepb.Store_SeriesServer) error {
+	if s.maxPendingRequests > 0 && s.pendingRequests.Load() >= s.maxPendingRequests {
```

Comment:
do we only limit on `Series` grpc, how about `RemoteWrite` ?

Reply:
`RemoteWrite` is limited in `pkg/receive/limiter.go`. `Series()` and `RemoteWrite()` handlers each have their own limiter, store.Limiter and receive.Limiter. They are limited separately for now, but the next step is to put them together so that when load is high, `Series()` requests are limited first before `RemoteWrite()` requests are limited.